src/Microsoft.AspNet.Cors.Core/CorsResult.cs

```diff
@@ -0,0 +1,179 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.AspNet.Cors.Core
+{
+    /// <summary>
+    /// Results returned by <see cref="ICorsEngine"/>.
+    /// </summary>
+    public class CorsResult
+    {
+        private long? _preflightMaxAge;
+
+        /// <summary>
+        /// Gets a value indicating whether the result is valid.
+        /// </summary>
+        public bool IsValid
+        {
+            get
+            {
+                return ErrorMessages.Count == 0;
+            }
+        }
+
+        /// <summary>
+        /// Gets the error messages.
+        /// </summary>
+        public IList<string> ErrorMessages { get; } = new List<string>();
```

Comment:
Simple POST with form data are prevented because CSRF token is missing, not because CORS processing is comparing policy. With simple requests stopping requests which don't match CORS policy doesn't really prevent attacks. As I understand it, the javascript would just as easily add a script src or create and submit form tag in that case to cause the request to be sent without the Origin header, bypassing the CORS policy anyway. The real feature is that if the javascript attempts any XHR or ajax call to get the data from the server, the browser won't share the results with the calling javascript.